### ISSUE TITLE
Is it neccessary that UnitfulStrings outlive the recipe?

### ISSUE THREAD

**Issue by BeastyBlacksmith** [jw3126/UnitfulRecipes.jl#74]
I am asking because the pgfplots backend does some processing on labels etc.  depending on whether its a `LaTeXString` or not.
But since `UnitfulString{LaTeXString, ...}` is not a `LaTeXString`, it goes the wrong path.

I could add extra handling for this, but in general I'd prefer if recipes didn't change the type of inputs.

**Comment by jw3126** [comment]
Not sure. What I imagine could happen is that when constructing a plot in multiple steps, `plot(...); plot!(...); plot!(...)` units get added multiple times.

**Comment by gustaphe** [comment]
Yes-ish. We need to keep track of both the unit and the manual part of the label separately so that we can

* `plot!` correctly
* change the axis unit after the fact (actually checking now we don't actually do this ...)

Perhaps there's another option, but it gets pretty tricky. What does pgfplots need to know it for?

**Comment by BeastyBlacksmith** [comment]
> What does pgfplots need to know it for?

It does escape certain characters, e.g. if you want to label something "100%".

**Comment by gustaphe** [comment]
Edit dumb. ~~Is that something you want? I would expect `%` in a LaTeXString to be the comment char. `UnitfulLatexify` for instance gives~~

~~But maybe the specific example is beside the point.~~

 If you have a good suggestion of how else we could convey and store this information I'd love to hear it. Otherwise maybe a guide interface somehow?
